packages/core/test/proxy-scripts/lsp-broker/LspUniswapV2Broker.js

```diff
@@ -496,4 +491,191 @@ contract("LspUniswapV2Broker", function (accounts) {
+      await lpToken.approve(router.address, MAX_UINT_VAL, { from: trader });
+
+      await router.removeLiquidity(
+        longToken.address,
+        shortToken.address,
+        (await lpToken.balanceOf(trader)).toString(),
+        "0",
+        "0",
+        trader,
+        MAX_UINT_VAL,
+        { from: trader }
+      );
+
+      // Trader should get back the exact number of long and short tokens with no slippage on either side.
+      assert.equal((await lpToken.balanceOf(trader)).toString(), toWei("0"));
+      assert.equal((await longToken.balanceOf(trader)).toString(), toWei("1000"));
+      assert.equal((await shortToken.balanceOf(trader)).toString(), toWei("1000"));
+    });
+  });
+  describe("atomicMintAddLiquidity: AMM contains Long against Short token in unequal ratio", () => {
+    beforeEach(async () => {
+      // Add 100000 long and 10000 short. this makes the price 100000/10000 long/short. In other words every short is
+      // worth 10 long. When LPing we will be market selling longs for shorts to reach the appropriate ratio.
+      await router.addLiquidity(
+        longToken.address,
+        shortToken.address,
+        toWei("100000"),
+        toWei("10000"),
+        "0",
+        "0",
+        deployer,
+        MAX_UINT_VAL,
+        { from: deployer }
+      );
+      lpToken = await Token.at(pair.address);
+
+      assert.equal((await longToken.balanceOf(pair.address)).toString(), toWei("100000"));
+      assert.equal((await shortToken.balanceOf(pair.address)).toString(), toWei("10000"));
+      assert.equal(await getPoolSpotPrice(longToken, shortToken), "10.0000"); // price should be exactly 10 TokenA/TokenB
+
+      // Mint EOA some collateral:
+      await collateralToken.mint(trader, toWei("1000"), { from: deployer });
+      assert.equal((await collateralToken.balanceOf(trader)).toString(), toWei("1000"));
+      await collateralToken.approve(lspUniswapV2Broker.address, toWei("1000"), { from: trader });
+    });
+
+    it("Can correctly mint and LP in one transaction", async function () {
+      // Calculate how many LP tokens are expected to be minted per unit collateral. Trader added 100000 units of
+      // to mint 100000 long and short tokens.
+      // so
+      const lpTokensPerCollateral = (await lpToken.balanceOf(deployer))
+        .mul(toBN(toWei("1"))) // scalding factor
+        .div(toBN(toWei("100000"))) // number of collateral units used in minting
+        .addn(1); // offset to fix rounding error
+
+      await lspUniswapV2Broker.atomicMintAddLiquidity(
+        true, // tradingAsEOA. true as calling from an EOA (not DSProxy).
+        longShortPair.address, // longShortPair. address to mint tokens against.
+        router.address, // router. uniswap v2 router to execute trades
+        toWei("1000"), // collateralToMintWith. we will use 1000 units of collateral to mint 1000 long and 1000 short tokens.
```

Comment:
can you try one test where the amount to mint is > 50% of the pool, to test what happens with the low level `UniswapPair.mint` call when slippage is really high? So say minting with 5,000 collateral? or even 50,000 collateral? (I don't know if that would even work?)